### ISSUE TITLE
[BUG] MediaElement Crash On IOS

### ISSUE THREAD

**Issue by Kremed** [CommunityToolkit/Maui#1010]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

I'm trying to implement Media Element on IOS, for some reason the app is stopped/disabled, the script works fine on Android

`Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<AVPlayerViewController: 0x1167e4800> should have parent view controller:<Microsoft_Maui_Platform_PageViewController: 0x113a15d50> but actual parent is:<Microsoft_Maui_Controls_Platform_ModalWrapper: 0x117ec2850>'`

**Android Platform :** 

https://user-images.githubusercontent.com/43956454/219947958-3ec0597b-00f6-4dd9-ac08-022e23162dfc.mp4

**IOS Platform :** 

https://user-images.githubusercontent.com/43956454/219948120-d4fd028c-2f19-402f-9b2e-57511eefeb6f.MP4

**Output Logs:** 

```

Resolved pending breakpoint at 'AuthIntrodactionView.xaml.cs:9,1' to void Mobile.AuthPages.Views.AuthIntrodactionView..ctor () [0x00007].
Resolved pending breakpoint at 'AuthIntrodactionViewModel.cs:36,1' to void Mobile.AuthPages.ViewModels.AuthIntrodactionViewModel..ctor () [0x0001a].
Resolved pending breakpoint at 'AuthIntrodactionViewModel.cs:39,1' to void Mobile.AuthPages.ViewModels.AuthIntrodactionViewModel..ctor () [0x00026].
The thread 0x6 has exited with code 0 (0x0).
The thread 0xd has exited with code 0 (0x0).
Thread finished: <Thread Pool> #6
Thread finished: <Thread Pool> #13
Thread started: <Thread Pool> #19
2023-02-19 13:31:35.382 Xamarin.PreBuilt.iOS[383:5841] *** Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<AVPlayerViewController: 0x1167e4800> should have parent view controller:<Microsoft_Maui_Platform_PageViewController: 0x113a15d50> but actual parent is:<Microsoft_Maui_Controls_Platform_ModalWrapper: 0x117ec2850>'
*** First throw call stack:
(0x1b14a9a34 0x1aa9c9958 0x1b158e34c 0x1b35f904c 0x1b35f8e78 0x1b3565228 0x1b3565264 0x1b3565264 0x1b3565264 0x1b3565264 0x1b356d934 0x1b37e796c 0x1b37e6f00 0x1b3cdba28 0x1b373e5d4 0x1b37e69ec 0x1b3730ad4 0x1b37309b4 0x1b3c4f708 0x1b36e1c24 0x1b36e1b60 0x1b83b7850 0x1b83b87c8 0x1b8399f18 0x1b8399b90 0x1b1533828 0x1b15174a8 0x1b151bb58 0x1e771b984 0x1b38c6628 0x1b38c62a0 0x1022dc5a4 0x1025f5aa0 0x1025f47c0 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250ecc0 0x1024cfa30 0x1024d8e74 0x1025f5a6c 0x1025f47f8 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250e618 0x1025bdf74 0x1023069bc 0x10262
4104 0x1ce1e5df0)

Resolved pending breakpoint at 'AuthIntrodactionView.xaml.cs:9,1' to void Mobile.AuthPages.Views.AuthIntrodactionView..ctor () [0x00007].
Resolved pending breakpoint at 'AuthIntrodactionViewModel.cs:36,1' to void Mobile.AuthPages.ViewModels.AuthIntrodactionViewModel..ctor () [0x0001a].
Resolved pending breakpoint at 'AuthIntrodactionViewModel.cs:39,1' to void Mobile.AuthPages.ViewModels.AuthIntrodactionViewModel..ctor () [0x00026].
The thread 0x6 has exited with code 0 (0x0).
The thread 0xd has exited with code 0 (0x0).
Thread finished: <Thread Pool> #6
Thread finished: <Thread Pool> #13
Thread started: <Thread Pool> #19
2023-02-19 13:31:35.382 Xamarin.PreBuilt.iOS[383:5841] *** Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<AVPlayerViewController: 0x1167e4800> should have parent view controller:<Microsoft_Maui_Platform_PageViewController: 0x113a15d50> but actual parent is:<Microsoft_Maui_Controls_Platform_ModalWrapper: 0x117ec2850>'
*** First throw call stack:
(0x1b14a9a34 0x1aa9c9958 0x1b158e34c 0x1b35f904c 0x1b35f8e78 0x1b3565228 0x1b3565264 0x1b3565264 0x1b3565264 0x1b3565264 0x1b356d934 0x1b37e796c 0x1b37e6f00 0x1b3cdba28 0x1b373e5d4 0x1b37e69ec 0x1b3730ad4 0x1b37309b4 0x1b3c4f708 0x1b36e1c24 0x1b36e1b60 0x1b83b7850 0x1b83b87c8 0x1b8399f18 0x1b8399b90 0x1b1533828 0x1b15174a8 0x1b151bb58 0x1e771b984 0x1b38c6628 0x1b38c62a0 0x1022dc5a4 0x1025f5aa0 0x1025f47c0 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250ecc0 0x1024cfa30 0x1024d8e74 0x1025f5a6c 0x1025f47f8 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250e618 0x1025bdf74 0x1023069bc 0x10262
4104 0x1ce1e5df0)

2023-02-19 13:31:35.383 Xamarin.PreBuilt.iOS[383:5841] *** Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<AVPlayerViewController: 0x1167e4800> should have parent view controller:<Microsoft_Maui_Platform_PageViewController: 0x113a15d50> but actual parent is:<Microsoft_Maui_Controls_Platform_ModalWrapper: 0x117ec2850>'
*** First throw call stack:
(0x1b14a9a34 0x1aa9c9958 0x1b158e34c 0x1b35f904c 0x1b35f8e78 0x1b3565228 0x1b3565264 0x1b3565264 0x1b3565264 0x1b3565264 0x1b356d934 0x1b37e796c 0x1b37e6f00 0x1b3cdba28 0x1b373e5d4 0x1b37e69ec 0x1b3730ad4 0x1b37309b4 0x1b3c4f708 0x1b36e1c24 0x1b36e1b60 0x1b83b7850 0x1b83b87c8 0x1b8399f18 0x1b8399b90 0x1b1533828 0x1b15174a8 0x1b151bb58 0x1e771b984 0x1b38c6628 0x1b38c62a0 0x1022dc5a4 0x1025f5aa0 0x1025f47c0 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250ecc0 0x1024cfa30 0x1024d8e74 0x1025f5a6c 0x1025f47f8 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250e618 0x1025bdf74 0x1023069bc 0x10262
4104 0x1ce1e5df0)
2023-02-19 13:31:35.383 Xamarin.PreBuilt.iOS[383:5841] *** Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<AVPlayerViewController: 0x1167e4800> should have parent view controller:<Microsoft_Maui_Platform_PageViewController: 0x113a15d50> but actual parent is:<Microsoft_Maui_Controls_Platform_ModalWrapper: 0x117ec2850>'
*** First throw call stack:
(0x1b14a9a34 0x1aa9c9958 0x1b158e34c 0x1b35f904c 0x1b35f8e78 0x1b3565228 0x1b3565264 0x1b3565264 0x1b3565264 0x1b3565264 0x1b356d934 0x1b37e796c 0x1b37e6f00 0x1b3cdba28 0x1b373e5d4 0x1b37e69ec 0x1b3730ad4 0x1b37309b4 0x1b3c4f708 0x1b36e1c24 0x1b36e1b60 0x1b83b7850 0x1b83b87c8 0x1b8399f18 0x1b8399b90 0x1b1533828 0x1b15174a8 0x1b151bb58 0x1e771b984 0x1b38c6628 0x1b38c62a0 0x1022dc5a4 0x1025f5aa0 0x1025f47c0 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250ecc0 0x1024cfa30 0x1024d8e74 0x1025f5a6c 0x1025f47f8 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250e618 0x1025bdf74 0
x1023069bc 0x102624104 0x1ce1e5df0)
2023-02-19 13:31:35.384 Xamarin.PreBuilt.iOS[383:5841] *** Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<AVPlayerViewController: 0x1167e4800> should have parent view controller:<Microsoft_Maui_Platform_PageViewController: 0x113a15d50> but actual parent is:<Microsoft_Maui_Controls_Platform_ModalWrapper: 0x117ec2850>'
*** First throw call stack:
(0x1b14a9a34 0x1aa9c9958 0x1b158e34c 0x1b35f904c 0x1b35f8e78 0x1b3565228 0x1b3565264 0x1b3565264 0x1b3565264 0x1b3565264 0x1b356d934 0x1b37e796c 0x1b37e6f00 0x1b3cdba28 0x1b373e5d4 0x1b37e69ec 0x1b3730ad4 0x1b37309b4 0x1b3c4f708 0x1b36e1c24 0x1b36e1b60 0x1b83b7850 0x1b83b87c8 0x1b8399f18 0x1b8399b90 0x1b1533828 0x1b15174a8 0x1b151bb58 0x1e771b984 0x1b38c6628 0x1b38c62a0 0x1022dc5a4 0x1025f5aa0 0x1025f47c0 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250ecc0 0x1024cfa30 0x1024d8e74 0x1025f5a6c 0x1025f47f8 0x1025ea19c 0x1025e7d70 0x1025b8d78 0x102507d70 0x10250
e618 0x1025bdf74 0x1023069bc 0x102624104 0x1ce1e5df0)

```

### Expected Behavior

It should appear as on the Android device, the video should be visible in the background

### Steps To Reproduce

- This Issue in an existing project, I will create a repository as soon as possible.

### Link to public reproduction project repository

- This Issue in an existing project, I will create a repository as soon as possible.

### Environment

```markdown
- .NET MAUI CommunityToolkit : 4.0.0
- .NET MAUI CommunityToolkit MediaElement: 1.0.1
- OS: IPhone X IOS 16.2 
- .NET MAUI :
```

**Comment by Kremed** [comment]
@brminnick  "needs reproduction" label Does this mean that you need more details from me?
Or will the problem be fixed in future releases by the community?

BTW I created a completely new app, and MediaElement worked as expected in iOS !? any instructions?

**Comment by brminnick** [comment]
From the "Reporting a bug" section in our contribution guidelines: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

> Bug reproduction
> 
> We always request a reproduction sample, and that's not to make your life hard or anything like that... The reason of having a reproduction is to save us time to indentify and fix the bug. Create a new project; download the MCT NuGet; write UI; ViewModel; create a service; run it. As you can see it's a lot of timing consuming for us that we could spend fixing the actual issue. So PLEASE create a small reproduction project, upload it on GitHub or GitLab and paste the link in the issue.
> 
> We don't accept .zip files as reproduction samples, for security reasons. So if you send us a .zip file we will kindly ask you to upload it to GitHub or GitLab and share the link
> 
> And most important: Please, help us to help you ❤️

**Comment by jfversluis** [comment]
I'm pretty sure this is resolved in version 1.0.2, please try that and close the issue if that fixes it.

If not, please provide the requested reproduction sample. Thanks!

**Comment by Kremed** [comment]
Exactly, it worked as expected when updating to version 1.0.2.
But to be a loop, I created a separate app with 1.0.1 and it worked too 
Thank you for your support
